### PR TITLE
Rename point-mutations to hamming

### PR DESCRIPTION
**Pull request by PurityControl**
Have renamed point-mutations to hamming as per issue #90.
Have renamed the directory but not the file. The file has always been named dna.erl / dna_test.erl so this has not bee changed but could be renamed to hamming if required.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "bob",
     "sum-of-multiples",
     "strain",
-    "point-mutations",
+    "hamming",
```
Comment (NobbZ) [inline config.json]: Please move the old `"point-mutations"` to the `"deprecated"` section at the bottom of the file `config.json` as well.

**Comment by NobbZ** [comment]
I'm fine with the changes as they are, but there's a small comment inlined.

**Comment by PurityControl** [comment]
thanks for the feedback will push the changes shortly

**Comment by kytrinyx** [comment]
I don't think GitHub notifies people if you make changes, so to be safe I'd recommend always adding a comment in the PR when you've made changes.

**Comment by PurityControl** [comment]
Yes, sorry about that. I didn't think that through properly. I'll commit and comment in the proper order next time.

**Pull request merged**